### PR TITLE
update version to 0.18.0

### LINKED ISSUE #3669: Release policy must be updated to bump the minor version of the main branch immediately following a release.
Structs used in NMR interface of native modules are not versioned. Instead, they rely on the version of bpf2c that was used to generate the native module. eBPF runtime is expected to use this version to handle multiple versions of native module.

To allow this, the version in the `main` branch should be bumped to next version once a version has been released.

We should also update the release process document to capture this step after a release is done.

### PR DESCRIPTION
**Pull request by shankarseal**
## Description

Fixes #3669 .
Updates the release policy to update the product version immediately after a release branch is forked. Updating the version to 0.18.0.

## Testing
CI.

## Documentation

No documentation changes.

## Installation

No setup changes.

### CODE REVIEW

File: docs/ReleaseProcess.md [hunk 1]
```diff
@@ -14,12 +14,12 @@ An issue with the title `Scheduled eBPF release is due` is automatically created
 version incremented every time. When this issue is triaged, a decision must be taken by the maintainers on whether to go ahead with the new monthly release. If
 the decision is to create a new release the release manager must proceed with the following process.
 1. Create a topic branch from "`main`" on your private repo fork, and check it out (e.g., `<user>/release-X-Y`).
-1. Follow the process in the [Updating the Release Version](ReleaseProcess.md#updating-the-release-version) to update the release version.
 1. Create a pull-request from your topic branch into the `main` branch of the [original "upstream" `ebpf-for-windows` repo]([https://github.com/microsoft/ebpf-for-windows), with the title of the PR as *"Release v`X.Y.0`"* (replace "`X`" and
-"`Y`" with the version number being released).
+"`Y`" with the curent version number, that is being released).
```
Comment (dthaler) [inline docs/ReleaseProcess.md:18]:
```suggestion
"`Y`" with the current version number that is being released).
```

**Review by dthaler — APPROVED** [review]
Ok modulo typos in the doc

File: resource/ebpf_version.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: MIT
 
 #define EBPF_VERSION_MAJOR 0
-#define EBPF_VERSION_MINOR 17
+#define EBPF_VERSION_MINOR 18
```
Comment (shpalani) [inline resource/ebpf_version.h:5]: Is this PR targeting #3683 for the next release?

Comment (shankarseal) [reply resource/ebpf_version.h:5]: No. This PR changes the release policy to update the version of the main branch post forking of the release branch.

File: docs/ReleaseProcess.md [hunk 3]
```diff
@@ -14,12 +14,12 @@ An issue with the title `Scheduled eBPF release is due` is automatically created
...
 1. Once the PR is approved and merged into the "`main`" branch, of the original `ebpf-for-windows` repo, create a new release branch from `main` from the
 **previous PR's commit**, and name it "`release/X.Y`".
-1. **IMPORTANT:** Once the release branch is created, no new feature work is allowed to be merged into that branch. However, bug fixes can be taken as deemed necessary by the
+1. Follow the process in the [Updating the Product Version](ReleaseProcess.md#updating-the-product-version) to update the version of the product in the main branch, for the next release.
+1. **IMPORTANT:** Once the release branch is created, no new feature work is allowed to be merged into the release branch. However, bug fixes can be taken as deemed necessary by the
```
[the remark this answers is not in the document]
Comment (shankarseal) [reply docs/ReleaseProcess.md]: Clarified and simplified the process. Thanks.

**Review by shpalani — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Review by mtfriesen — APPROVED** [review]

**Pull request merged**